### ISSUE TITLE
twisted.test.test_application.TestInternet2.testUDP doesn't test UDPServer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2942]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2942|
|Type|defect|
|Created|2007-12-09 21:19:44Z|
|Branch|https://github.com/twisted/twisted/tree/test-udp-2942|

It tests TCPServer.

<details><summary>Searchable metadata</summary>

```
trac-id__2942 2942
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_test_udp_2942 branches/test-udp-2942
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1197235184000000 1197235184000000
changetime__1198882405000000 1198882405000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [22089]) Branching to 'test-udp-2942'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [22090]) Fix testUDP and add a docstring.

Refs [#2942](https://github.com/twisted/twisted/issues/2942)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

This is ready to review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @therve|
|-|-|

Looks good

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [22176]) Merge test-udp-2942

Author: therve
Reviewer: exarkun
Fixes [#2942](https://github.com/twisted/twisted/issues/2942)

Fix twisted.test.test_application.TestInternet2.testUDP to actually test
UDPServer instead of TCPServer.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
